auth/oci.go

```diff
@@ -0,0 +1,99 @@
+// Copyright 2017 CoreOS, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package auth
+
+import (
+	"crypto/rsa"
+	"crypto/x509"
+	"encoding/pem"
+	"fmt"
+	"io/ioutil"
+	"os/user"
+	"path/filepath"
+	"strings"
+
+	"github.com/go-ini/ini"
+)
+
+const OCIConfigPath = ".oraclebmc/mantle"
+
+// OCIProfile represents a parsed OCI profile.
+type OCIProfile struct {
+	TenancyID   string `ini:"tenancy"`
+	UserID      string `ini:"user"`
+	Fingerprint string `ini:"fingerprint"`
+	KeyFile     string `ini:"key_file"`
+	Region      string `ini:"region"`
+}
+
+// ReadOCIConfig decodes an OCI config file, for more information:
+// https://docs.us-phoenix-1.oraclecloud.com/Content/API/Concepts/sdkconfig.htm
+//
+// If path is empty, $HOME/.oraclebmc/mantle is read.
```

Comment:
The thinking was that we'd read the existing `config` file for keys in the official schema, and `mantle` for additional keys like `compartment`.  I guess that means the profile struct will need to have keys from both files.